src/features/ExtensionCommands.ts

```diff
@@ -388,44 +389,106 @@ export class ExtensionCommandsFeature implements IFeature {
+                    }
+                    newFileAbsolutePath = path.join(workspaceRootUri.fsPath, saveFileDetails.newPath);
+                }
+
+                await this.saveDocumentContentToAbsolutePath(currentFileUri, newFileAbsolutePath);
+                return EditorOperationResponse.Completed;
+
+            default:
+                // Other URI schemes are not supported
+                const msg = JSON.stringify(saveFileDetails);
+                this.log.writeVerbose(
+                    `<ExtensionCommands>: Saving a document with scheme '${currentFileUri.scheme}' ` +
+                    `is currently unsupported. Message: '${msg}'`);
+                return EditorOperationResponse.Completed;
         }
+    }
 
-        // Otherwise we want to save as a new file
-
-        // First turn the path we were given into an absolute path
-        // Relative paths are interpreted as relative to the original file
-        const newFileAbsolutePath = path.isAbsolute(saveFileDetails.newPath) ?
-            saveFileDetails.newPath :
-            path.resolve(path.dirname(saveFileDetails.filePath), saveFileDetails.newPath);
-
-        // Retrieve the text out of the current document
-        const oldDocument = await vscode.workspace.openTextDocument(saveFileDetails.filePath);
-
-        // Write it to the new document path
-        fs.writeFileSync(newFileAbsolutePath, oldDocument.getText());
-
-        // Finally open the new document
-        const newFileUri = vscode.Uri.file(newFileAbsolutePath);
-        const newFile = await vscode.workspace.openTextDocument(newFileUri);
-        vscode.window.showTextDocument(newFile, { preview: true });
+    /**
+     * Take a document available to vscode at the given URI and save it to the given absolute path
+     * @param documentUri the URI of the vscode document to save
+     * @param destinationAbsolutePath the absolute path to save the document contents to
+     */
+    private async saveDocumentContentToAbsolutePath(
+        documentUri: vscode.Uri,
+        destinationAbsolutePath: string): Promise<void> {
+            // Retrieve the text out of the current document
+            const oldDocument = await vscode.workspace.openTextDocument(documentUri);
+
+            // Write it to the new document path
+            try {
+                // TODO: Change this to be asyncronous
+                fs.writeFileSync(destinationAbsolutePath, oldDocument.getText());
+            } catch (e) {
+                this.log.writeAndShowWarning("<ExtensionCommands>: " +
+                    `Unable to save file to path '${destinationAbsolutePath}': ${e}`);
+                return;
+            }
 
-        return EditorOperationResponse.Completed;
-   }
+            // Finally open the new document
+            const newFileUri = vscode.Uri.file(destinationAbsolutePath);
+            const newFile = await vscode.workspace.openTextDocument(newFileUri);
```

Comment:
Yea!  Glad to see some async/await make its way into the extension. I find it easier to read than a bunch of `.then()` continuations.  That's probably the C# dev in me coming out but I think TS/JS devs will like it too.

Reply:
Yes, @SeeminglyScience set me on the right path there after I tried to put in a particularly twisted chain of `.then()`s when I first made this feature.